src/compile/axis.ts

```diff
@@ -137,7 +138,7 @@ export function title(model: Model, channel: Channel) {
   }
 
   // if not defined, automatically determine axis title from field def
-  const fieldTitle = model.fieldTitle(channel);
+  const fieldTitle = fieldDefTitle(model.fieldDef(channel), axis.showUnit === 'title' || !axis.showUnit);
```

Comment:
Right, I think I added it later because it wasn't working. I will take another look.